src/ProjectTemplates/Directory.Build.targets

```diff
@@ -0,0 +1,120 @@
+<Project>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.targets', '$(MSBuildThisFileDirectory)../'))" />
+
+  <!--
+    Template projects use <ProjectReference> items to:
+
+    1. Ensure `build.cmd -vs AI` or `build.cmd -vs ProjectTemplates`
+       results in the referenced project being included in the SDK.sln
+       file, along with its transitive project references.
+
+    2. Collect the Package ID and Version from the project for use in
+       the template using the "GetProjectPackageVersion" target defined
+       in /eng/MSBuild/Packaging.targets.
+
+    This <ItemDefinitionGroup> must be defined in targets after
+    Microsoft.Common.CurrentVersion.Targets  so that it can override the
+    <ItemDefinitionGroup> defined there.
+
+    - ReferenceOutputAssembly - don't pass to the compiler
+    - Targets - use this target instead of the default GetTargetPath
+    - SetTargetFramework - only a single framework build is needed
```

Comment:
Are you sure you need `SetTargetFramework`?  I think you could evaluate the project without setting any value for `TargetFramework`.  This is an 'outer build' and is safer than setting global-properties which might cause another evaluation.  What made you add SetTargetFramework?  If you found it was to avoid leaking properties a better option would be to use `UndefineProperties`.